src/backend/distributed/executor/multi_utility.c

```diff
@@ -677,8 +679,9 @@ PlanIndexStmt(IndexStmt *createIndexStatement, const char *createIndexCommand)
 			{
 				DDLJob *ddlJob = palloc0(sizeof(DDLJob));
 				ddlJob->targetRelationId = relationId;
+				ddlJob->preventTransaction = createIndexStatement->concurrent;
 				ddlJob->commandString = createIndexCommand;
-				ddlJob->taskList = DDLTaskList(relationId, createIndexCommand);
+				ddlJob->taskList = IndexTaskList(relationId, createIndexStatement);
```

Comment:
Would it be better to continue using worker_apply_shard_ddl_command() for regular index commands given that worker_apply_shard_ddl_command() is more native(?) and more tested?

```C
				if (createIndexStatement->concurrent)
				{
					ddlJob->taskList = IndexTaskList(relationId, createIndexStatement);
				} else {
					ddlJob->taskList = DDLTaskList(relationId, createIndexCommand);
				}
```